src/utl/src/ScopedTemporaryFile.cpp

```diff
@@ -33,4 +36,70 @@ ScopedTemporaryFile::~ScopedTemporaryFile()
   }
 }
 
+StreamHandler::StreamHandler(const char* filename, bool binary)
+    : filename_(filename)
+{
+  std::string tmp_filename = filename_ + ".tmp";
+  if (fs::exists(tmp_filename)) {
+    fs::remove(tmp_filename);
+  }
+  os_.exceptions(std::ofstream::failbit | std::ofstream::badbit);
+  try {
+    if (binary) {
+      os_.open(tmp_filename, std::ios::binary | std::ios_base::out);
+    } else {
+      os_.open(tmp_filename, std::ios_base::out);
+    }
+  } catch (std::ios_base::failure& e) {
+    std::string error = e.what();
```

Comment:
I don't want to loose the stack trace by replacing the exception.

Please try with https://en.cppreference.com/w/cpp/error/throw_with_nested and confirm that it works:

```
  } catch (const std::ios_base::failure& e) {
    std::string error = e.what();
    std::throw_with_nested(std::ios_base::failure(error + " (failed to open '" + tmp_filename + "')"));
  }
}

// Helper function to print nested exceptions
void printNestedException(const std::exception& e, int level = 0) {
  std::cerr << std::string(level, ' ') << "Exception: " << e.what() << std::endl;
  try {
    std::rethrow_if_nested(e);
  } catch (const std::exception& nested) {
    printNestedException(nested, level + 1);
  } catch (...) {
    // No further nested exception
  }
}

int main() {
  try {
    StreamHandler sh("example.txt", true);
  } catch (const std::exception& e) {
    printNestedException(e);
  }
  return 0;
}
```

Reply:
I don't think that we loose stack trace in this case as both nested exception and the original one have the same stack trace. Anyway do you prefer `std::throw_with_nested`  ?

Reply:
> I don't think that we loose stack trace in this case as both nested exception and the original one have the same stack trace. Anyway do you prefer `std::throw_with_nested` ?

You are right. To be a bit nit-picky: we do loose a bit of stack information, but nothing interesting: inside libstdc++ when opening a file, not application stack trace.

However, it would be nice to have an example of std::throw_with_nested, so it is a "bonus point" for this PR, but not a concern for this PR as such.

Up to you.